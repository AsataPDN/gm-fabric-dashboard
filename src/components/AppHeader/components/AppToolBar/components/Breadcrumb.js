import styled from "styled-components";

import { contrastColor, spacingScale } from "style/styleFunctions";
import { COLOR_ALT_BACKGROUND } from "style/styleVariables";
import { media } from "style/styleVariables";

const APP_TOOLBAR_BACKGROUND_COLOR = COLOR_ALT_BACKGROUND.string();

const Breadcrumb = styled.li`
  flex: 0 0 auto;
  display: flex;
  align-items: center;
  color: black;
  ${media.breadcrumbsBreakpoint200`
    max-width: calc(100vw/4);
    overflow: hidden;
  `} ${media.breadcrumbsBreakpointHandheld`
    max-width: calc(100vw/9);
    overflow: hidden;
  `} &:before {
    content: ">";
    transform: scaleX(0.5);
    display: flex;
    color: ${contrastColor(APP_TOOLBAR_BACKGROUND_COLOR, 0.8).string()};
    padding: 0 ${spacingScale(0.5)};
  }
  a {
    display: flex;
    max-width: auto;
    white-space: nowrap;
    text-overflow: ellipsis;
    color: inherit;
    padding: ${spacingScale(1)} 0;
    color: ${contrastColor(APP_TOOLBAR_BACKGROUND_COLOR, 0.85).string()};
    &:hover {
      color: ${contrastColor(APP_TOOLBAR_BACKGROUND_COLOR, 1).string()};
    }
  }

  &:first-child {
    &:before {
      content: none;
    }
    a {
      padding-left: ${spacingScale(2)};
    }
  }
`;

export default Breadcrumb;
