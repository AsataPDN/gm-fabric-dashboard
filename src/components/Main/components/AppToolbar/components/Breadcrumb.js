import styled from "styled-components";

const Breadcrumb = styled.li`
  flex: 0 0 auto;
  display: flex;
  align-items: center;
  color: black;
  &:before {
    content: ">";
    transform: scaleX(0.5);
    display: flex;
    opacity: 0.5;
    padding: 0 4px;
  }
  a {
    display: flex;
    max-width: 100%;
    white-space: nowrap;
    text-overflow: ellipsis;
    padding: 8px 0;
    color: inherit;
    &:hover {
      color: black;
    }
  }
  &:first-child {
    &:before {
      content: none;
    }
    a {
      padding-left: 16px;
    }
  }
`;

export default Breadcrumb;
