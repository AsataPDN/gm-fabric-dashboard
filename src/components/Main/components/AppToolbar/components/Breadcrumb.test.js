import React from "react";
import { shallow } from "enzyme";
import Breadcrumb from "./Breadcrumb";

describe("Breadcrumb", () => {
  it("matches snapshot", () => {
    const aBreadcrumb = shallow(<Breadcrumb />);
    expect(aBreadcrumb).toMatchSnapshot();
  });
});
