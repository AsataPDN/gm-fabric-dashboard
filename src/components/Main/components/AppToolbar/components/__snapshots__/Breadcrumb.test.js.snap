// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumb matches snapshot 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: black;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
}

.c0:before {
  content: ">";
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  opacity: 0.5;
  padding: 0 4px;
  -webkit-transform: scaleX(0.5);
  -ms-transform: scaleX(0.5);
  transform: scaleX(0.5);
}

.c0 a {
  color: inherit;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  max-width: 100%;
  padding: 8px 0;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.c0 a:hover {
  color: black;
}

.c0:first-child:before {
  content: none;
}

.c0:first-child a {
  padding-left: 16px;
}

<li
  className="c0"
/>
`;
